x/ibc-hooks/types/errors.go

```diff
@@ -1,8 +1,14 @@
 package types
 
+import sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
 var (
-	ErrBadPacketMetadataMsg = "cannot unmarshal metadata: '%v'. %s"
 	ErrBadMetadataFormatMsg = "wasm metadata not properly formatted for: '%v'. %s"
 	ErrBadExecutionMsg      = "cannot execute contract: %v"
-	ErrBadResponse          = "cannot create response: %v"
+
+	ErrMsgValidation = sdkerrors.Register("wasm-hooks", 1, "error in wasmhook message validation")
```

Comment:
Just curious why use "wasm-hooks" instead of ModuleName? Also is it possible in this case to avoid using sdkerrors? AFAIK our goal in general is to refrain from using sdk errors

Reply:
> Just curious why use "wasm-hooks" instead of ModuleName? 

This error is specific to the validation of the wasm hook. The module could (and probably will in the future) have different hooks. That said, don't mind changing it.

> is it possible in this case to avoid using sdkerrors? AFAIK our goal in general is to refrain from using sdk errors

What's the alternative? Ibc internally expects an error with codespace and code id. (https://github.com/cosmos/ibc-go/blob/b40dbc646c7191aaee12d3a70243f9b6f3f80919/modules/core/04-channel/types/acknowledgement.go#L32-L42)

Reply:
> This error is specific to the validation of the wasm hook. The module could (and probably will in the future) have different hooks. That said, don't mind changing it.
Ah I see what you were trying to do. LGTM!

> What's the alternative? Ibc internally expects an error with codespace and code id. (https://github.com/cosmos/ibc-go/blob/b40dbc646c7191aaee12d3a70243f9b6f3f80919/modules/core/04-channel/types/acknowledgement.go#L32-L42)

Don't know much about ibc-go but is this the method that receives the error? https://github.com/cosmos/ibc-go/blob/b40dbc646c7191aaee12d3a70243f9b6f3f80919/modules/core/04-channel/types/acknowledgement.go#L32-L35 looks like it takes in go lang error. Or I assume there might be different components in ibc-go that accepts codespace and code-id. If so, please feel free to resolve this comment!

Reply:
yeah, we can pass a golang errors (the sdk errors are golang errors under the hood). I guess I could define the error type using `sdkerrors.New` or manually as `Error{codespace: codespace, code: code, desc: desc}`, but not sure what the advantage of doing that would be.

Maybe we should have a standard way to do this across osmosis?

Reply:
@ValarDragon can you chime in here? I think we should leave these using sdkerrors. AFAICT, passing a normal error to IBC's `channeltypes.NewErrorAcknowledgement` will return the default "internalABCICode" (1). Having the code numbers is useful to determine what happened.

Reply:
Agreed, SDK error for this makes sense

(It doesn't for most other modules tho)

Reply:
Hrmm, actually no, I'm onboard with just doing normal golang errors. I don't think people are actually going to parse off these codespaces, vs just looking at events / explorers to diagnose error.

Reply:
We're generally trying to rethink codespacing in cosmos: https://github.com/cosmos/cosmos-sdk/issues/14070

Very few people actually use codespacing in their integrator logic. 

I'm not very opinionated on this, since its unlikely these errors are things we'd want to change in a patch release.

Reply:
I believe we have to start registration from 2 onwards though

Reply:
Ah given that this is IBC-standard here, agreed with SDK registering these

Reply:
ok, so, in conclusion: keep sdk errors for this so it matches other IBC errors and move the first code to 2 (which was just an oversight on my part), right?

Reply:
yup!